src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -64,35 +68,42 @@ public boolean execute(final CompileContext context) {
       // delegate dependencies resolution to rebar
       return true;
     }
+    LOG.info("Prepare build order for project "+project.getName());
 
     File projectSystemDirectory = BuildManager.getInstance().getProjectSystemDirectory(project);
     if (projectSystemDirectory == null) {
       addPrepareDependenciesFailedMessage(context);
       return true;
     }
-
-    ErlangModuleBuildOrders buildOrders = ApplicationManager.getApplication().runReadAction(new Computable<ErlangModuleBuildOrders>() {
+    LOG.debug("Run action for preparing build order");
+    ErlangProjectBuildOrder projectBuildOrder = ApplicationManager.getApplication().runReadAction(new Computable<ErlangProjectBuildOrder>() {
       @Nullable
       @Override
-      public ErlangModuleBuildOrders compute() {
-        return getModuleBuildOrders(context);
+      public ErlangProjectBuildOrder compute() {
+        return getProjectBuildOrder(context);
       }
     });
-    if (buildOrders == null) {
+    if (projectBuildOrder == null) {
       return false; // errors are reported to context.
     }
     try {
-      Document serializedBuildOrders = new Document(XmlSerializer.serialize(buildOrders, new SkipDefaultValuesSerializationFilters()));
-      File file = new File(projectSystemDirectory, ErlangBuilder.DEPENDENCIES_CONFIG_FILE_PATH);
+      LOG.debug("Serialize build order");
+      Document serializedDocument = new Document(XmlSerializer.serialize(projectBuildOrder, new SkipDefaultValuesSerializationFilters()));
+      File file = new File(projectSystemDirectory, ErlangPrepareDependenciesBuilder.DEPENDENCIES_TREE_FILE_PATH);
       //noinspection ResultOfMethodCallIgnored
       file.getParentFile().mkdirs();
-      JDOMUtil.writeDocument(serializedBuildOrders, file, SystemProperties.getLineSeparator());
+      LOG.debug("Write build order to "+file.getAbsolutePath());
+      JDOMUtil.writeDocument(serializedDocument, file, SystemProperties.getLineSeparator());
     }
     catch (XmlSerializationException e) {
+      LOG.warn("Can't serialize build order object.");
+      LOG.debug(e);
       addPrepareDependenciesFailedMessage(context);
       return true;
     }
     catch (IOException e) {
+      LOG.warn("Some I/O errors occurred while writing build orders to file");
+      LOG.debug(e);
```

Comment:
So, you only log exceptions in debug level, while you log useless warning messages in warn level. Why?

Also, serialization failure means a programming error - there has to be something wrong with objects you serialize - log it as an error.